Proposal: Removing sudo from examples

moxiegirl opened the pull request:
**Situation**
Currently, we propose users set up Docker and sudo rules such that depending on configuration sudo is required . For Mac users on Boot2Docker we actually historically told them not to use sudo.

Removing sudo from our documentation is the most inclusive way to document. 
- For those Mac etc users and users without sudo requirements, no harm no foul. 
- For those in restrictive environment, well they will need to use `sudo` but they will be used to that anyway not just for issuing docker commands but for other third-party commands which I am sure are similarly restricted. If these users forget, the note is there to remind them if not the fact the command fails with a perm error.

**Proposal**
We now have instructions in our Unix installs about setting up docker group to avoid sudo.  So, let us sudo from the examples and any other file. Let's add a boilerplate section when appropriate reminding them that if they have to use `sudo` to run docker they can change that.  I propose to do this in all our docs.  If the proposal is accepted; I'll do the sweep.  run.md attached as an example.

Signed-off-by: Mary Anthony mary@docker.com

Commit:
Removing sudo from examples

We now have instructions in our Unix installs about setting up
docker group to avoid sudo.  Also, Mac/Windows shouldn't use
sudo.  So, I've removed sudo from our examples and added a
section at the top reminding them that if they have to use
sudo to run docker they can change that.

Signed-off-by: Mary Anthony <mary@docker.com>

moxiegirl commented:
@SvenDowideit @jamtur01 @fredlf @spf13

cgwalters commented:
See also https://github.com/docker/docker/pull/10144 which goes the opposite direction.

moxiegirl commented:
@cgwalters I don't quite follow your reference or is it inference. :-D  We've actually *already* removed `sudo` from other parts of the documentation. The result is that we have inconsistent inclusion of `sudo` in our examples.  

The reality is currently we allow users to set up Docker and sudo rules such that  _depending on configuration_ sudo is required .  For Mac users on Boot2Docker we actually historically told them not to use sudo. 

Removing sudo is the most inclusive way to document. For those Mac etc users **and** users without `sudo requirements, no harm no foul. For those in restrictive environment, well they will need to use `sudo` but they will be used to that anyway not just for issuing `docker` commands but for other third-party commands which I am sure are similarly restricted.  If these users forget, the note is there to remind them if not the fact the command fails with a perm error.

@jfrazelle  let me know if I'm wildly off the reservation; always possible.

jamtur01 commented on docs/sources/reference/run.md:
```diff
@@ -24,26 +24,30 @@ other `docker` command.
+The `run` options control the image's runtime behavior in a container. These
+settings affect:
+
+ * detached or foreground running
```
I like periods at the end of bulleted lists but others MMV

jamtur01 commented:
Assuming everyone is in agreement then it LGTM.

phemmer commented:
See #8056 which this is effectively the opposite of.

moxiegirl commented:
@phemmer thanks for pointer.  That was merged in Sept 2014. I'll get folks in a pod and talk.

fredlf commented:
The goal should be to have examples that most users can copy/paste in most situations. I honestly cannot say whether that means including `sudo` or not. My instinct says not, but I would like to discuss with some people to get more data. 

It would be nice if we (docs writers) could discuss large-scale changes like this as a group to make sure we are serving the needs of all users.

moxiegirl commented:
@fredlf I've only made changes to this single file.  Unfortunately, I was unaware there was any controversy over sudo or not sudo.  I had seen an early PR sweep that Jess did removing sudo from examples; also I did not include sudo in the contributors guide; so, unfortunately, I assumed the sudo in the doc were legacy left overs from a sweep.

When I saw the inclusion/exclusion of `sudo` raised in a bday contributor's submission; I merged his and opened this.  My thinking was, yes, we needed a larger discussion.  I certainly think the writers should discuss it but I also think any discussion should include the the maintainers as a whole.  This is, in part, a technical issue.  My current understanding is that PRs are the good forum for raising and discussing these issues with maintainers.  Point taken, though, I'll reframe this as a proposal though...to clarify it as such.

fredlf commented:
Awesome, thanks. A proposal seems like the best way to go about this. 

FWIW, my informal polling this morning of support, training and SE folk is leaning pretty hard towards the approach you have here: remove `sudo` from examples, but be sure to include a note about using it as needed.

moxiegirl commented:
Solomon brought @naelmohammad (a user at the party) stopped by the docs desk to give this feedback "You need to remove the sudo from your examples; it really caused problems for my friend and me tonight."

fredlf commented:
Yeah, I talked to a bunch of people throughout yesterday and the consensus was pretty clear, we should remove it.

moxiegirl commented:
Good. Glad you feel comfortable with this now Fred.  Would you like to add an LGTM?

fredlf commented:
Oh, right. LGTM

crosbymichael commented:
LGTM

Commit:
Removing sudo from command line examples

We now advise people to configure docker group and
add to sudo.  Mac shouldn't use sudo.  Removed sudo
from command examples.  Left in installation to be removed
in installation doc sweep -- removing requires finer
grain control.

Signed-off-by: Mary Anthony <mary@docker.com>

moxiegirl commented:
@jamtur01 @SvenDowideit @fredlf @spf13 one more set of eyes before I commit these changes please.  Thanks for your help everyone!

jamtur01 commented:
LGTM
